iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2217,7 +2219,7 @@ static int SendMqttConnectMsg(PMQTTTRANSPORT_HANDLE_DATA transport_data)
 
     if (result == 0)
     {
-        if (!transport_data->isProductInfoSet)
+        if (!transport_data->isOptionalParameterSet)
         {
```

Comment:
>{ [](start = 8, length = 1)

The block between the { and } here is up to +50 lines.  Please create a static helper function to do this work here.

Reply:
Also could we see about not having the if/nesting, e.g.

if (doSomething1 == error) 
{
   error
}
else 
{
   if (doSomething2 == error) 
   {
     error
   }
   else 
    {
         ... continuing to go deeper

and move to the more standard

if (doSomething1 == error) 
{
   error
 }
else if (doSomething2 == error) 
{
   error
}

---
In reply to: [385295048](https://github.com/Azure/azure-iot-sdk-c/pull/1439#discussion_r385295048) [](ancestors = 385295048)